server/src/main/java/io/crate/execution/engine/indexing/ShardingUpsertExecutor.java

```diff
@@ -254,15 +256,33 @@ private boolean shouldPauseOnPartitionCreation(ShardedRequests<ShardUpsertReques
         return false;
     }
 
+    static long reserveBytes(CircuitBreaker circuitBreaker) {
+        long minAcceptableBytes = ByteSizeUnit.MB.toBytes(5);
+        long free = circuitBreaker.getFree();
+        long wantedBytes = (long) (free * BREAKER_LIMIT_PERCENTAGE);
```

Comment:
If free is > 5 MB, but  free * 0.15 isn't, we should probably still allow it?

Reply:
👍  good point